duckduckgo/Android
# Remove all deps form DuckDuckGoApplication

Task/Issue URL: https://app.asana.com/0/414730916066338/1200695027776651/f
Tech Design URL: 
CC: 

**Description**:
The `DuckDuckGoApplication` class injects a lot of dependencies just because they need to execute some code in app `onCreate`. 
We introduced a while back a [plugin point](https://github.com/duckduckgo/Android/pull/1228) that enabled any class that wanted to execute anything during the app lifecycle do it without the need for the `DuckDuckGoAppliation` to know about it. But didn't migrated ALL possible deps at the time.

This PR is the second attempt

This PR:
- Move loadTrackerData() to a lifeccycle obserber plugin
- Removed unused settingsDataStore
- Move notificationRegistrar to use lifecycle observer
- Move httpsUpgrader to use lifecycle observer
- Move AndroidWorkScheduler to use lifecycle observer
- Tidy up DI for JobCleaner
- Move ShortcutReceiver to use lifecycle observer
- Move AppDataLoader to use licecycle observer
- Move AtbInitializer to use lifecycle observer
- Move OfflinePixelScheduler to use lifecycle observer


Still left TODO
* Move `submitUnsentFirePixels` to use the plugin point
* Move the `initialiseReferralRetrieval()` to use the plugin point

**Steps to test this PR**:
1. Install from this branch
1. Launch the app
1. Verify "Loading tracker data"  timber message appears
1. Verify "loading surrogate data"  timber message appears
1. Verify "Reload Https upgrader data" timber message appears
1. Verify "Started to load app data" and "Finished loading app data" appears
1. Verify "Scheduling offline pixels to be sent" appears
1. Verify "Scheduling work" timber message appears
1. Home the app and come back to it
1. Verify "Scheduling work" timber message appears
1. Verify "Initialize ATB" appears
1. Home the app and re-open it
1. Verify "Initialize ATB" appears on resume
1. Navigate to google.com
1. Go to overflow menu and "Add to him"
1. Verify shortcut has been added to home
1. Home the app
1. Go to overflow menu and "Add to home"
1. Verify another shortcut has been added to home
1. Close and re-open the app
1. Go to overflow menu and "Add to home"
1. Verify shortcut has been added to home


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
